stdlib/public/core/Sequence.swift

```diff
@@ -1124,18 +922,21 @@ extension Sequence where SubSequence == AnySequence<Element> {
     }
 
     if i != ringBuffer.startIndex {
-      let s0 = ringBuffer[i..<ringBuffer.endIndex]
-      let s1 = ringBuffer[0..<i]
-      return AnySequence([s0, s1].joined())
+      var rotated: [Element] = []
+      rotated.reserveCapacity(ringBuffer.count)
+      rotated += ringBuffer[i..<ringBuffer.endIndex]
+      rotated += ringBuffer[0..<i]
+      return rotated
```

Comment:
How does this copying approach compare to in-place rotation?
````swift
ringBuffer[0..<i].reverse()
ringBuffer[i..<ringBuffer.endIndex].reverse()
ringBuffer[0..<ringBuffer.endIndex].reverse()
````

Reply:
Also, is there a potential perf. win in using `ContiguousArray`s here (and in `dropLast`) and then `return Array(ringBuffer)`? I've noticed that's what `map` and `filter` implementations use.

Reply:
Either are worth exploring to see – they're not important for the ABI though so we can look at trying them out after this change lands.